### ISSUE TITLE
calling multiple populate() with same key returns empty objects

### ISSUE THREAD

**Issue by tsterker** [balderdashy/waterline#522]
I have some code where I dynamically create a query and ran into the following issue:

Assuming a user model:
```javascript
// User.js
module.exports = {
    attributes: {
        name: 'string',
        stats: {
            model: 'Stats',
            via: 'user'
        }
    }
};
```

and the following query (assuming a user with `id` 1):
```javascript
User.find(1).populate('stats').populate('stats').then(console.log);
```
yields a `user` with empty stats object:
```javascript
{
    name: 'dude',
    stats: { }
}
```
Note that only using one populate yields the user's stats.

In my case it was not too big of a deal to keep track of the populates and prevent the same populate key to occur twice. But I guess it is still unintended behavior.

**Comment by nkibler7** [comment]
I'm going to revive this. I'm having the same issue, looking into it now.

**Comment by sailsbot** [comment]
Thanks for posting, @tsterker.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
